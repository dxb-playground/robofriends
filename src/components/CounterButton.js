import React, { PureComponent } from 'react';

class CounterButton extends PureComponent {
  constructor() {
    super();
    this.state = {
      count: 0,
    };
  }

  // shouldComponentUpdate(nextProps, nextState) {
  //   return this.state.count !== nextState.count;
  // }

  render() {
    return (
      <button color={this.props.color} onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }

  updateCount = () => {
    this.setState(state => ({ count: state.count + 1 }));
  }
}

export default CounterButton;
