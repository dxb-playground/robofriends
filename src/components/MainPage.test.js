import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';
import MainPage from './MainPage';

describe('MainPage', () => {
  let wrapper;

  beforeEach(() => {
    let mock_props = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false,
    };
    wrapper = shallow(<MainPage {...mock_props} />);
  });

  it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('filters robots correctly', () => {
    let mock_props = {
      onRequestRobots: jest.fn(),
      robots: [
        {
          id: 3,
          name: 'Jon',
          email: 'jon@gmail.com',
        }
      ],
      searchField: 'a',
      isPending: false,
    };

    let wrapper = shallow(<MainPage {...mock_props} />);

    expect(wrapper.instance().filterRobots([])).toEqual([]);
  });
});
