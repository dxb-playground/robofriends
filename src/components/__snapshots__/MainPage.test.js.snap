// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders MainPage without crashing 1`] = `ShallowWrapper {}`;
